vllm/core/block/block_table.py

```diff
@@ -105,12 +108,27 @@ def append_token_ids(self,
         Args:
             token_ids (List[int]): The sequence of token IDs to be appended.
         """
-        assert self._is_allocated
+        assert self._is_allocated, "no blocks have been allocated"
         assert len(self._blocks) > 0
 
+        if self._block_sliding_window is not None:
+            null_block = self._allocator.null_block
+            if num_computed_slots is None:
+                num_computed_slots = self._num_full_slots
+            end_idx = (num_computed_slots //
+                       self._block_size) - self._block_sliding_window
+            for idx in range(0, end_idx):
+                b = self._blocks[idx]
+                if b is not null_block:
+                    self._allocator.free(b)
+                    self._blocks[idx] = null_block
```

Comment:
nit: I suggest moving this to its own class `SlidingWindowBlockTable` so the complexity is separated

Reply:
When I tried that, I had to repeat the whole body of the fork() method; now this is not big deal, but if someone adds a field to BlockTable, they may forget to update the other fork in SlidingWindowBlockTable. I'm not sure it's worth it. WDYT?